src/main/java/luffy/Deadline.java

```diff
@@ -0,0 +1,32 @@
+package luffy;
+
+/**
+ * Deadline class to represent a deadline.
+ *
+ * @author Silas Tay A0233425M
+ */
+public class Deadline extends Task{
+
+    /**
+     * Constructor for Deadline task.
+     *
+     * @param description Description of Deadline
+     * @param by Deadline for deadline task
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.type = "[D]";
+        this.by = by;
```

Comment:
Any reason why you decided to put the `by` attribute as part of `Task` class? Perhaps it will be better to have it as an attribute of `Deadline` class as it is only applicable to `Deadline` class and not other class that inherit from `Task`. Same goes for other fields such as `at`.